wily/__main__.py

```diff
@@ -223,8 +239,14 @@ def report(ctx, file, metrics, number, message, format, console_format, output):
     default=True,
     help="Show function/class level metrics where available",
 )
+@click.option(
+    "--thresholds",
+    type=ThresholdsParamType(),
```

Comment:
Are you sure you pass an instance here and not a type?

Reply:
[yep](https://click.palletsprojects.com/en/7.x/parameters/#implementing-custom-types). I also added a test for this.